docs/architecture/adr-018-extendable-voting-period.md

```diff
@@ -0,0 +1,66 @@
+
+So for example, if the `MaxVotingPeriodExtension` is set to 100 Days, then anyone with 1% of voting power can extend the voting power by 1 day.  If 33% of voting power has sent the message, the voting period will be extended by 33 days.  Thus, if absolutely everyone chooses to extend the voting period, the absolute maximum voting period will be `MinVotingPeriod + MaxVotingPeriodExtension`.
+
+This system acts as a sort of distributed coordination, where individual stakers choosing to extend or not, allows the system the guage the conentiousness/complexity of the proposal.  It is extremely unlikely that many stakers will choose to extend at the exact same time, it allows stakers to view how long others have already extended thus far, to decide whether or not to extend further.
+
+### Dealing with Unbonding/Redelegation
+
+There is one thing that needs to be addressed.  How to deal with redelegation/unbonding during the voting period.  If a staker of 5% calls `MsgExtendVotingPeriod` and then unbonds, does the voting period then decrease by 5 days again?  This is not good as it can give people a false sense of how long they have to make their decision.  For this reason, we want to design it such that the voting period length can only be extended, not shortened.  To do this, the current extension amount is based on the highest percent that voted extension at any time.  This is best explained by example:
+
+1. Let's say 2 stakers of voting power 4% and 3% respectively vote to extend.  The voting period will be extended by 7 days.
+2. Now the staker of 3% decides to unbond before the end of the voting period.  The voting period extension remains 7 days.
+3. Now, let's say another staker of 2% voting power decides to extend voting period.  There is now 6% of active voting power choosing the extend.  The voting power remains 7 days.
+4. If a fourth staker of 10% chooses to extend now, there is a total of 16% of active voting power wishing to extend.  The voting period will be extended to 16 days.
+
+### Delegators
+
+Just like votes in the actual voting period, delegators automatically inherit the extension of their validators.  If their validator chooses to extend, their voting power will be used in the validator's extension.  However, the delegator is unable to override their validator and "unextend" as that would contradict the "voting power length can only be ratcheted up" principle described in the previous section.  However, a delegator may choose the extend using their personal voting power, if their validator has not done so.
+
+## Status
+
+Proposed
+
+## Consequences
+
+### Positive
+
+- More complex/contentious governance proposals will have more time to properly digest and deliberate
+
+### Negative
+
+- Governance process becomes more complex and requires more understanding to interact with effectively
+- Can no longer predict when a governance proposal will end. Can't assume order in which governance proposals will end.
```

Comment:
I think this is a pretty substantial negative consequence - predicting when e.g. upgrade proposals will end is quite important for scheduling & preparation.

Reply:
For upgrade proposals, I think we should be using relative dates rather than absolute anyways.  i.e. "7 days after the block on which the voting period ends"

Reply:
The other option is to allow the proposer to make a proposal unextendable.  But good social norms against that unless necessary.